pkg/ccl/backupccl/backup_job.go

```diff
@@ -230,8 +231,22 @@ func backup(
 					TotalEntryCounts:  backupManifest.EntryCounts,
 					RevisionStartTime: backupManifest.RevisionStartTime,
 				})
+
+				// Always write checkpoints to the progress folder, but only write
+				// to base directory if we detect that the cluster isn't running
+				// on pure 22.1. This is because older nodes expect checkpoints
+				// in the base directory and will throw an error if it isn't.
+				// TODO (Darryl): Figure out if this is the right way to check for pure 22.1
+				if !settings.Version.IsActive(ctx, clusterversion.Start22_1) {
+					err := writeBackupManifest(
+						ctx, settings, defaultStore, backupManifestCheckpointName, encryption, backupManifest,
+					)
+					if err != nil {
+						log.Errorf(ctx, "unable to checkpoint backup descriptor: %+v", err)
+					}
+				}
 				err := writeBackupManifest(
-					ctx, settings, defaultStore, backupManifestCheckpointName, encryption, backupManifest,
+					ctx, settings, defaultStore, progressDirectory+"/"+backupManifestCheckpointName, encryption, backupManifest,
```

Comment:
ditto about `progressDirectory+"/"+backupManifestCheckpointName` being a const.

Reply:
I made some helper functions for reading/writing and now progressDirectory+"/"+... is being used way less but still a few times since it can be either progressDirectory+"/"+checkpoint or progressDirectory+"/"+temp-checkpoint. I could make a const for both if you think it's worth it for readability though